### ISSUE TITLE
Add support for zstd compression

### ISSUE THREAD

**Issue by probonopd** [AppImageCommunity/libappimage#193]
We are trying to standardize on zstd (no pun intended). So if libappiage doesn't support it yet, it would be highly appreciated if someone could add it.

@philmb3487 over at https://github.com/AppImage/appimagetool/issues/68#issuecomment-2365232487:

Some people are generating AppImages with `zstd` compression, and that is not supported out of the box by libappimage.
Now, that is fine for executing AppImages, but file thumbnailers such as KDE's Dolphin use libappimage.so directly to peek at the .DirIcon information, which fails for `zstd` images.

Would it be possible to add this support?

Arch Linux seems to enable fuse to get that support, squashfuse then needs to be compiled with `zstd` support turned on. That's a lot of switches that might or might not be enabled, depending on the distribution, and I think thumbnails should always be available in any case.

See my findings here,
https://github.com/ivan-hc/Steam-appimage/issues/27#issuecomment-2365229961

So, I propose either,

1) better zstd support out of the box,
2) document what needs to be done if Thumbnails cannot be shown on *some* appimages (out of the box it would be fine with xz, zip images,

```
Squashfs image uses (null) compression, this version supports only xz, zlib.
ERROR: appimage_read_file_into_buffer_following_symlinks : sqfs_open_image error: /home/phil/Applications/Steam-1.0.0.81-2-3-x86_64.appimage
```

**Comment by probonopd** [comment]
There is a pull request that did not get merged so far, maybe it can be reactivated:

* https://github.com/AppImageCommunity/libappimage/pull/158

**Comment by probonopd** [comment]
Currently we are using https://github.com/vasi/squashfuse/commit/1f98030 from 2016. We should probably use a more recent version.

**Comment by TheAssassin** [comment]
Closed by #196.
